### PR TITLE
📖 Update BootstrapConfig contract to include pausing

### PR DESCRIPTION
**Pull request by theobarberbany**
What this PR does / why we need it:
This change updates the BootstrapConfig provider contract to account for a new paused
condition.

Relates https://github.com/kubernetes-sigs/cluster-api/issues/10130

/area documentation

### CODE REVIEW

**Review by vincepri — APPROVED** [review]
/lgtm
/assign @fabriziopandini

File: docs/book/src/developer/providers/contracts/bootstrap-config.md [hunk 1]
```diff
@@ -283,6 +284,12 @@ the implication of this choice which are described both in the document above an
 
 </aside>
 
+### BootstrapConfig: pausing
+
+Providers SHOULD implement the pause behaviour for every object with a reconciliation loop. This is done by checking if `spec.paused` is set on the Cluster object and by checking for the `cluster.x-k8s.io/paused` annotation on the BootstrapConfig object.
```
Comment (sbueringer) [inline docs/book/src/developer/providers/contracts/bootstrap-config.md:289]: Unrelated to this PR, but I just found a mistake in a previous PR

> Providers SHOULD implement the pause behaviour for every object with a reconciliation loop. This is done by checking if `spec.paused` is set on the Machine object and by checking for the `cluster.x-k8s.io/paused` annotation on the InfraMachine object.

checking if `spec.paused` is set on the Machine object

should be

checking if `spec.paused` is set on the Cluster object

Comment (theobarberbany) [reply docs/book/src/developer/providers/contracts/bootstrap-config.md:289]: 🤦 Good catch! I'll update it :)

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**